libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -177,48 +178,112 @@ - (void)userAccountManager:(SFUserAccountManager *)userAccountManager
     if (nil != delegate) {
         request.delegate = delegate;
     }
-    
     [self.activeRequests addObject:request];
 
     // If there are no demonstrable auth credentials, login before sending.
-    CSFNetwork *currentNetwork = self.currentNetwork;
-    SFUserAccount *user = currentNetwork.account;
+    SFUserAccount *user = [SFUserAccountManager sharedInstance].currentUser;
+    __weak __typeof(self) weakSelf = self;
     if (user.credentials.accessToken == nil && user.credentials.refreshToken == nil && request.requiresAuthentication) {
-        [self log:SFLogLevelInfo msg:@"No auth credentials found.  Authenticating before sending request."];
+        [self log:SFLogLevelInfo msg:@"No auth credentials found. Authenticating before sending request."];
         [[SFAuthenticationManager sharedManager] loginWithCompletion:^(SFOAuthInfo *authInfo) {
-            [request prepareRequestForSend];
-            [currentNetwork executeAction:request.action];
+            __strong typeof(weakSelf) strongSelf = weakSelf;
+            [strongSelf enqueueRequest:request delegate:delegate];
         } failure:^(SFOAuthInfo *authInfo, NSError *error) {
-            [self log:SFLogLevelError format:@"Authentication failed in SFRestAPI: %@.  Logging out.", error];
+            [self log:SFLogLevelError format:@"Authentication failed in SFRestAPI: %@. Logging out.", error];
             NSMutableDictionary *attributes = [[NSMutableDictionary alloc] init];
             attributes[@"errorCode"] = [NSNumber numberWithInteger:error.code];
             attributes[@"errorDescription"] = error.localizedDescription;
             [SFSDKEventBuilderHelper createAndStoreEvent:@"userLogout" userAccount:nil className:NSStringFromClass([self class]) attributes:attributes];
             [[SFAuthenticationManager sharedManager] logout];
         }];
     } else {
-        // Auth credentials exist.  Just send the request.
-        [request prepareRequestForSend];
-        [currentNetwork executeAction:request.action];
+
+        // Auth credentials exist. Just send the request.
+        [self enqueueRequest:request delegate:delegate];
+    }
+}
+
+- (void)enqueueRequest:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate {
+    __weak __typeof(self) weakSelf = self;
+    NSURLRequest *finalRequest = [request prepareRequestForSend];
+    if (finalRequest) {
+        SFNetwork *network = [[SFNetwork alloc] init];
+        NSURLSessionDataTask *dataTask = [network sendRequest:finalRequest dataResponseBlock:^(NSData *data, NSURLResponse *response, NSError *error) {
+            __strong typeof(weakSelf) strongSelf = weakSelf;
+            if (error) {
+                [strongSelf log:SFLogLevelDebug format:@"REST request failed with error: Error Code: %ld, Description: %@, URL: %@", (long) error.code, error.localizedDescription, finalRequest.URL];
+
+                // Checks if the request was canceled.
+                if (error.code == -999) {
```

Comment:
Calling the appropriate delegate callbacks here to keep API behavior the same as before.